docs/DeveloperGuide.md

```diff
@@ -257,329 +257,368 @@ _{Explain here how the data archiving feature will be implemented}_
+    
     Use case ends.
 
 **Extensions**
 
-* 2a. The list is empty.
+* 1a. The list is empty
   
   Use case ends.
+
+* 2a. The given index is invalid
+
+  * 2a1. TAB shows an error message
+    
+    Use case resumes at step 1.
   
+* 2b. TAB detects that the entered lesson data does not satisfy requirements
 
-* 3a. The given index of the person is invalid.
-    * 3a1. TAB shows an error message.
+    * 2b1. TAB displays an error message and the recommended changes
 
-* 3b. The given index of the lesson of the person is invalid.
-    * 3b1. TAB shows an error message.
+      Use case resumes at step 1.
 
-* 3c. There is no specified field to edit.
-    * 3c1. TAB shows an error message.
-  
-  Use case resumes from step 2.
+<br/>
 
-**Use case: UC7 – Find student by student fields**
+#### Use case: UC5 – Edit Student’s Lesson Information
 
-**MSS:**
+**MSS**
 
-1.  User requests to list students.
-2.  TAB shows a list of students.
-3.  User requests to find student with a specified name and address.
-4.  TAB shows a list of students with matching name and address.
+1. User [<ins>gets a list of students (UC1)</ins>](Use-case:-UC1---Get-a-list-of-students)
+2. User requests to edit certain fields of a specific lesson of a specific student in the list
+3. TAB updates the corresponding fields of the lesson for the specified student
 
-    Use case ends.
+   Use case ends.
 
-**Extensions:**
+**Extensions**
 
-* 2a. The list is empty.
+* 1a. The list is empty
 
   Use case ends.
 
-**Use case: UC8 – Filter students by tags**
+* 2a. The specified student has no lessons
 
-**MSS:**
+  Use case ends.
 
-1.  User request to view all tags.
-2.  TAB shows a list of created tags.
-3.  User requests to filter students by a few specified tags.
-4.  TAB shows a list of students matching the specified tags.
+* 2b. The given index of the student is invalid
 
-    Use case ends.
+    * 2b1. TAB displays an error message
 
-**Extensions:**
+      Use case resumes at step 1.
 
-* 2a. There are no tags.
+* 2c. The given index of the lesson of the student is invalid
 
-  Use case ends.
+    * 2b1. TAB displays an error message
 
-**Use Case: UC8 – See quick start instructions**
+      Use case resumes at step 1.
 
-**MSS:**
-1. User opens the app.
-2. TAB shows quick tips in the command result box.
-   Use case ends.
-   
-**Use Case: UC9 – See help guide**
+* 2b. No lesson fields to be edited are specified
 
-**MSS:**
+    * 2b1. TAB displays an error message
 
-1. User requests for help.
-2. TAB shows a summary table of all the command usages as well as a link to user guide page. 
+      Use case resumes at step 1.
 
-   Use case ends.
+* 2d. TAB detects that the entered lesson data does not satisfy requirements
+
+    * 2d1. TAB displays an error message and the recommended changes
 
-**Use Case: Add a Student**
+      Use case resumes at step 1.
+
+* 2e. TAB detects clashing lessons
+
+    * 2e1. TAB displays an error message
+
```

Comment:
Can I clarify if this is a feature we would be adding or do you mean TAB detects duplicate lessons?

Reply:
Yup this is the duplicate lessons feature, except i think clashing lessons is a more correct description of what that feature is

Reply:
From my understanding, clashing lessons can mean that the time range overlaps each other, rather than being "equal", is that similar to your view?

Reply:
Yep, overlapping time range